### PR TITLE
Handle connect_timeout

### PR DESCRIPTION
**Pull request by baelter**
### Friendly reminders
- [x] Describe the problem / feature (ideally with [meaningful commit messages](https://tekin.co.uk/2019/02/a-talk-about-revision-histories))
- [x] Lint rules pass
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] The environment (`heroku config`) has been updated if needed (new `ENV` variables)

### Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: lib/sparoid.rb [hunk 1]
```diff
@@ -48,6 +48,8 @@ def fdpass(ips, port, connect_timeout: 10) # rubocop:disable Metrics/AbcSize, Me
     # wait for any socket to be connected
     until sockets.empty?
       _, writeable, = IO.select(nil, sockets, nil, connect_timeout)
+      next if writeable.nil? # Timeout, try next socket
```
Comment (dentarg) [inline lib/sparoid.rb:51]: Can't we be stuck in this loop (`until sockets.empty?`) if all sockets timeout?

Comment (baelter) [reply lib/sparoid.rb:51]: Yes, look like it, but we don't know which socket timed out?

Comment (carlhoerberg) [reply lib/sparoid.rb:51]: yes, we do, there's an error array, the third array. tested this on linux but it was always empty, will fix

Comment (carlhoerberg) [reply lib/sparoid.rb:51]: no, actually, on timeout, it means that all sockets timed out, so: 
```
_, writeable, = IO.select(nil, sockets, nil, connect_timeout) || break
```
works

**Pull request merged**